File: 20260329/lc11.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let ans = 0,
    n = height.length;
  if (n < 2) {
    return ans;
  }

  let l = 0,
    r = n - 1;
  while (l < r) {
    let area = (r - l) * Math.min(height[l], height[r]);
    ans = Math.max(ans, area);
    if (height[l] < height[r]) {
      l++;
    } else {
      r--;
    }
  }
  return ans;
};

File: 20260329/lc11.cpp
#include <algorithm>
#include <vector>

using namespace std;

class Solution {
public:
  int maxArea(vector<int> &height) {
    if (height.size() < 2) {
      return 0;
    }
    int maxArea = 0;
    int left = 0, right = height.size() - 1;
    while (left < right) {
      int area = (right - left) * min(height[left], height[right]);
      maxArea = max(maxArea, area);
      if (height[left] < height[right]) {
        left++;
      } else {
        right--;
      }
    }
    return maxArea;
  }
};